filebeat/prospector/factory.go

```diff
@@ -9,13 +10,13 @@ import (
 
 // Factory is a factory for registrars
 type Factory struct {
-	outlet    Outlet
+	outlet    *channel.Outlet
```

Comment:
would be nice to still have an interface instead of passing `*channel.Outlet` all over the place. interfaces can be stubbed more easily for testing.